### PR TITLE
Set global NIO options.

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
This uses a snapshot version of gcloud, to get the setDefaultCloudStorageConfiguration feature.

Do not submit, because of the reliance of a self-compiled gcloud.

### CODE REVIEW

**Comment by jean-philippe-martin** [comment]
@droazen @lbergelson here's the code you were asking for. I'm not certain as to where's the best place to set global defaults; for now I put them in Main.

File: src/main/java/org/broadinstitute/hellbender/Main.java [hunk 1]
```diff
@@ -147,6 +148,7 @@ public Object instanceMain(final String[] args) {
      * Note: this is the only method that is allowed to call System.exit (because gatk tools may be run from test harness etc)
      */
     protected final void mainEntry(final String[] args) {
+        BucketUtils.setGlobalNIODefaultOptions();
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/Main.java]: Can you move to `CommandLineProgram.instanceMainPostParseArgs()`, just below the call to `BlockGunzipper.setDefaultInflaterFactory()`? Should also add logger output to `CommandLineProgram.printSettings()` to output the current default retries/reopens for GCS.

**Comment by droazen** [comment]
@jean-philippe-martin I get the following error when trying to use the shaded google-cloud-java snapshot jar:

```
/Users/droazen/.m2/repository/com/google/cloud/google-cloud-nio/0.20.2-alpha-SNAPSHOT/google-cloud-nio-0.20.2-alpha-SNAPSHOT-shaded.jar(com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.class): warning: Cannot find annotation method 'value()' in type 'AutoService': class file for com.google.auto.service.AutoService not found
```

Checking the jar, `AutoService` is indeed not present, but is an annotation on `CloudStorageFileSystemProvider`. Any thoughts on what could be causing this?

**Comment by jean-philippe-martin** [comment]
I'm looking into this right now.

**Comment by droazen** [comment]
Thanks @jean-philippe-martin. If there's some change I can make in my clone of `google-cloud-java` to get the shaded snapshot working, let me know!

Interestingly, the `AutoService` annotation is marked as `@Retention(SOURCE)`...

**Comment by jean-philippe-martin** [comment]
Marking `AuthService` as `optional` in gcloud-nio's pom fixes it:

```
<dependency>
    <groupId>com.google.auto.service</groupId>
    <artifactId>auto-service</artifactId>
    <version>1.0-rc3</version>
    <optional>true</optional>
    <scope>provided</scope> <!-- to leave out of the all-deps jar -->
</dependency>
```

$ ./gradlew sparkJar
BUILD SUCCESSFUL

$ ./gatk-launch ExampleNioCountReads (...)
(starts without error)

**Comment by droazen** [comment]
Thanks @jean-philippe-martin -- this worked, though it took me a while to realize that the update from `1.0-rc2` to `1.0-rc3` was also essential (still failed for me with `1.0-rc2`)

**Comment by jean-philippe-martin** [comment]
ah sorry @droazen , I should have mentioned it. I didn't test with 1.0-rc2.

**Comment by droazen** [comment]
Closing -- this was done in https://github.com/broadinstitute/gatk/pull/3295

**Pull request closed without merging**